File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/visitor/hosting/ServerlessService.java
// ServerlessService.java

package com.bigboxcode.designpattern.visitor.hosting;

public class ServerlessService implements Service {

    private final double hourlyPrice = 0.32;
    private int totalHours;

    public ServerlessService(int totalHours) {
        this.totalHours = totalHours;
    }

    public double getHourlyPrice() {
        return hourlyPrice;
    }

    public int getTotalHours() {
        return totalHours;
    }

    @Override
    public double accept(HostingCalculatorVisitor hostingCalculatorVisitor) {
        return hostingCalculatorVisitor.visit(this);
    }
}

File: design-patterns/ts/visitor/hosting-cost/serverless-service.ts
// serverless-service.ts

import HostingCalculatorVisitor from "./hosting-calculator-visitor";
import Service from "./service";

class ServerlessService implements Service {

    private readonly hourlyPrice: number = 0.32;
    private totalHours: number;

    constructor(totalHours: number) {
        this.totalHours = totalHours;
    }

    getHourlyPrice(): number {
        return this.hourlyPrice;
    }

    getTotalHours(): number {
        return this.totalHours;
    }

    accept( hostingCalculatorVisitor: HostingCalculatorVisitor): number {
        return hostingCalculatorVisitor.visit(this);
    }
}

export default ServerlessService;